packaging/ansible-runner-service-project/project/roles/ovirt-ova-extract/tasks/main.yml

```diff
@@ -1,15 +1,33 @@
 ---
+- name: Create temporary directory
+  tempfile:
+    state: directory
+    suffix: ova
+  register: ova_temp
+
+- name: Copy extract_ova.py to temp directory
+  copy:
+    src: extract_ova.py
+    dest: "{{ ova_temp.path }}/extract_ova.py"
+
 - name: Run extraction script
-  script: >
-    extract_ova.py
+  command: >
+    "{{ ansible_python_interpreter }}"
+    "{{ ova_temp.path }}/extract_ova.py"
     "{{ ovirt_import_ova_path }}"
     "{{ ovirt_import_ova_disks }}"
     "{{ ovirt_import_ova_image_mappings }}"
-  args:
-    executable: "{{ ansible_python_interpreter }}"
+  async: "{{ ansible_timeout }}"
+  poll: 15
   register: extraction_result
 
 - name: Check OVA extraction process result
   fail:
     msg: "Failed to extract OVA file"
   when: extraction_result.rc is defined and extraction_result.rc != 0
+
+always:
+  - name: Remove temp directory
+    file:
+      state: absent
+      path: "{{ ova_temp.path }}"
```

Comment:
To use `always` you need to put previous code to the `block`.
https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html#id3 
This goes for all changed files.

Reply:
Done, thanks